provider/maas/environ.go

```diff
@@ -261,7 +262,22 @@ func (env *maasEnviron) SetCloudSpec(_ stdcontext.Context, spec environscloudspe
 	}
 
 	apiVersion := apiVersion2
-	controller, err := GetMAASController(maasServer, maasOAuth)
+	args := gomaasapi.ControllerArgs{
+		BaseURL: maasServer,
+		APIKey:  maasOAuth,
+	}
+	// If the user has specified to skip TLS verification, we need to
+	// add a new http client with insecure TLS (skip verify).
+	if spec.SkipTLSVerify {
+		args.HTTPClient = &http.Client{
```

Comment:
Can you verify we honor the proxy changes, or do we need to inject the HTTP client that does?

Reply:
As discussed offline, this http client was nil before and gomaasapi created an empty client when it was nil, so no changes from this point of view. As per the proxy settings, I think that should be configured on MAAS itself so it can access their needed images.